PurchasesCoreSwift/FoundationExtensions/DateFormatter+Extensions.swift

```diff
@@ -0,0 +1,36 @@
+//
+// Created by Andrés Boedo on 7/29/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+extension DateFormatter {
+
+    static let iso8601SecondsDateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
+        return dateFormatter
+    }()
+
+    static func iso8601SecondsOrMillisecondsDate(fromString maybeDateString: String?) -> Date? {
```

Comment:
looking at this again, I think: 
since the method just returns a `Date`, which isn't format-specific, but _does_ work exclusively with ISO8601 strings, this should be named: 

```suggestion
    static func date(fromiso8601SecondsOrMillisecondsString maybeDateString: String?) -> Date? {
```

Reply:
are we always passing in an optional? just wondering whether it makes sense for it to take in an optional string

Reply:
looks like the only direct call is from the ArraySlice extensions class...i kinda like it as optional since that's what our other extension kinda "gets us". what are your thoughts?